### PR TITLE
Added `gat`, `gats`, `many gat`, `many gats` commands

### PR DESCRIPTION
**Pull request by ArtemIsmagilov**
implement this commands - `gat`, `gats`, `many_gat`, `many_gats`

### CODE REVIEW

File: emcache/client.py [hunk 1]
```diff
@@ -647,6 +651,79 @@ async def version(self, memcached_host_address: MemcachedHostAddress) -> Optiona
         _, version_value = result.split(b" ")
         return version_value.decode()
 
+    async def gat(self, key: bytes, exptime: int = 0, return_flags=False) -> Optional[Item]:
+        """Gat command is used to fetch item and update the
+        expiration time of an existing item.
+
+        gat <exptime> <key>\r\n
+        """
+        keys, values, flags, _ = await self._fetch_command(b"gat %d" % exptime, key)
```
Comment (pfreixes) [inline emcache/client.py:660]: reusing the `_fetch_command` and adding as a command the interpolation of the command and the exptime seems to twist a bit the interfaces. The other option would to also create a two new code paths for the fetch commands that require the `gat`(s)  commands. Yes I know will imply add more code but defenitly would look nicer :)

What do you think?

**Review by pfreixes — CHANGES_REQUESTED** [review]

File: emcache/base.py [hunk 2]
```diff
@@ -263,6 +263,22 @@ async def version(self, memcached_host_address: MemcachedHostAddress) -> Optiona
         server.
         """
 
+    @abstractmethod
+    async def gat(self, key: bytes, exptime: int = 0, return_flags=False) -> Optional[Item]:
```
[the remark this answers is not in the document]
Comment (ArtemIsmagilov) [reply emcache/base.py:267]: I agree, this option distorts the interface

File: emcache/client.py (hunk 1 above)
Comment (ArtemIsmagilov) [reply emcache/client.py:660]: thanks for the note, it’s really better not to mix everything into one team

Comment (ArtemIsmagilov) [reply emcache/client.py]: used new method _get_and_touch_command with required arg exptime

**Review by pfreixes — APPROVED** [review]
Awesome, thanks for the contribution!

**Pull request merged**